addons/purchase/views/purchase_views.xml

```diff
@@ -216,13 +216,15 @@
                                                 <label for="product_qty"/>
                                                 <div>
                                                     <field name="product_qty" class="oe_inline"/>
-                                                    <span class="oe_inline">&#160;</span>
-                                                    <field name="product_uom" groups="uom.group_uom" class="oe_inline"/>
+                                                    <field name="product_uom" groups="uom.group_uom" class="oe_inline ml-2"/>
                                                 </div>
+                                                <field name="qty_received_method" invisible="1"/>
+                                                <field name="qty_received" attrs="{'invisible': [('parent.state', 'not in', ('purchase', 'done'))], 'readonly': [('qty_received_method', '!=', 'manual')]}"/>
```

Comment:
I think you can remove the "readonly" part as this form is only opened in readonly mode.
That way you can also get rid of the "qty_received_method" field.

Reply:
@awa-odoo , what happens when someone removes the `editable="bottom"` from the tree view? Thanks

Reply:
@mba-odoo Then the commit that removes the editable will also have to check the fields to synchronize domain etc. but we want to keep the fields as simple as possible if it's readonly mode.

I actually asked @tde-banana-odoo before adding this comment and he agrees.

Reply:
@awa-odoo , But if someone change using web studio(mainly on saas) or directly remove that using debug mode then they probably don't know what should be the domain, context or read-only attributes. that's why I prefer to keep it.

Reply:
@mba-odoo I get your point, forget the comments linked to that matter.

We will just have to pay much attention to the fact that tree view and form view field attributes stay synchronized in the future then.